subprojects/gradle/cthulhu/src/main/kotlin/com/avito/plugin/MarkReportAsSourceAction.kt

```diff
@@ -0,0 +1,61 @@
+package com.avito.plugin
+
+import com.avito.logger.Logger
+import com.avito.report.ReportsApi
+import com.avito.report.model.GetReportResult
+import com.avito.report.model.ReportCoordinates
+import com.avito.time.TimeProvider
+import com.github.salomonbrys.kotson.jsonObject
+
+/**
+ * TMS will use reports marked with special analyzer_key as source of truth
+ *
+ * related tasks:
+ * CTHL-495
+ * MBS-6483
+ */
+class MarkReportAsSourceAction(
+    private val reportsApi: ReportsApi,
+    private val timeProvider: TimeProvider,
+    private val logger: Logger
+) {
+
+    fun mark(reportCoordinates: ReportCoordinates) {
+        val testSuiteVersion = timeProvider.nowInMillis()
+
+        logger.info("This is a new version [$testSuiteVersion] of full test suite for tms")
+
+        val reportId = tryGetId(reportCoordinates)
+
+        if (reportId != null) {
```

Comment:
I think it is a critical function of that plugin. Should we fail?

Reply:
good catch; yes we should act just as in any other fail see: https://github.com/avito-tech/avito-android/pull/567#discussion_r484354869